### ISSUE TITLE
P0: rewire PCA onto SVD + add TruncatedSVD — retires the only measured sklearn loss in BEATS.md

### ISSUE THREAD

**Issue by noahgift** [paiml/aprender#3148]
Parent: #3146 · Tier **P0 — substrate** · Milestone **0.68.0** · **Depends on: SVD ticket (T1)**

## The gap

`crates/aprender-core/src/preprocessing/pca.rs:334` computes PCA by forming the covariance matrix and eigendecomposing it:

```rust
// Convert to trueno Matrix for eigendecomposition
let eigen = SymmetricEigen::new(&cov_matrix)
```

This is the textbook-wrong route for PCA, for two independent reasons:

1. **Numerical.** Forming `X^T X` squares the condition number. Small singular values of `X` are lost to rounding in a way that SVD on the centered data does not suffer. On ill-conditioned feature matrices the trailing components are noise.
2. **Complexity.** It is O(d^3) in the number of FEATURES, so wide data (d >> n) is catastrophically slow — exactly the regime where randomized SVD wins.

This is not theoretical. `docs/BEATS.md` records it as the **only measured loss against sklearn**:

> **PCA fit_transform vs sklearn** — apr ~18.6x *slower* — sklearn delegates to LAPACK-SVD; apr's decomposition is unoptimized

That row is the most defensible criticism a linfa or sklearn user can make of aprender today.

## Scope

1. Rewire `PCA` to run SVD on the **centered** data matrix. Never form the covariance matrix.
2. Auto-select exact vs randomized SVD by problem shape (sklearn's `svd_solver='auto'` heuristic is a reasonable starting point; document whatever is chosen).
3. Add a `TruncatedSVD` estimator (SVD without centering — the correct transform for sparse/TF-IDF data, where centering destroys sparsity). This is what makes `TfidfVectorizer` usable for LSA.
4. Support `n_components` as an integer, as a float in (0,1) meaning explained-variance ratio, and expose `explained_variance_`, `explained_variance_ratio_`, `singular_values_`.

## Acceptance criteria (each must be able to FAIL)

- [ ] A falsification test asserts the covariance path is **gone**: `preprocessing/pca.rs` contains no construction of a `d x d` covariance matrix. A grep-based guard is acceptable only with an anchored pattern and a must-match/must-not-match case table.
- [ ] Ill-conditioning test: on a matrix with singular values spanning `1e0 .. 1e-9`, the recovered components match the SVD-exact components to `1e-7`, where the current covariance path is asserted to FAIL the same tolerance (proving the fix changed something).
- [ ] `explained_variance_ratio_` sums to 1.0 within 1e-12 when `n_components == min(n, d)`.
- [ ] Sign-convention determinism: two fits on the same data produce byte-identical components.
- [ ] Oracle parity vs `sklearn.decomposition.PCA` on pinned fixtures — components match up to sign within 1e-8.
- [ ] `TruncatedSVD` on a TF-IDF matrix from the existing `TfidfVectorizer` produces a reduced representation; a nearest-neighbour retrieval test on it beats random baseline by a margin asserted in the test.
- [ ] **The BEATS.md loss row is retired.** `beat-sklearn-pca-speed-v1.yaml` lands with the measured ratio, and `docs/BEATS.md` no longer lists PCA under "Speed loss". If the new ratio is still a loss, the row stays and says so honestly — but it must be re-measured, not deleted.
- [ ] `crates/aprender-core/tests/readme_contract.rs` still passes (it gates BEATS.md against contracts).

## Note on honesty

Do not assume this produces a win. The acceptance criterion is a **re-measurement**, not a beat. If apr is still slower than LAPACK after the rewrite, BEATS.md records the new number and the ticket still closes — the numerical correctness fix stands on its own.

**Comment by noahgift** [comment]
triage 2026-09-23 (aprender-dd): milestoned to 0.76.0, theme Classical ML + AutoML. It is the classical-ML parity work (#3146 epic, CLUS rows), which no 0.70–0.75 theme covered. Operator selection 2026-09-23 ("New 0.76 release theme", wording by the cop aprender-cf); moved out of `backlog`.
